app/models/user.rb

```diff
@@ -1,12 +1,23 @@
 class User < ActiveRecord::Base
+
   # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable and :omniauthable
-  devise :database_authenticatable, :registerable,
+  # :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable, :confirmable,
          :recoverable, :rememberable, :trackable, :validatable
 
+  ## Associations
   belongs_to :company
-  
   has_many :invoices, through: :company
   has_many :clients, through: :company
-  
+
+  ## Callbacks
+  after_create :assign_role_and_admin_id
+
+  def assign_role_and_admin_id
+    if (self.role.nil? && self.admin_id.nil?)
```

Comment:
the default context will always be `self`. So unless there is conflict between local variable and method, you should not use `self`